jitterdebugger.c

```diff
@@ -40,44 +40,49 @@
 #include <sys/syscall.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
+#include <stdbool.h>
 #include <fcntl.h>
 #include <linux/limits.h>
 
 #include "jitterdebugger.h"
 
+#define FMT_THREAD "T:%2u (%5lu) A:%2u C:%14" PRIu64
 #define VT100_ERASE_EOL		"\033[K"
 #define VT100_CURSOR_UP		"\033[%uA"
 
-#define NSEC_PER_SEC		1000000000
-#define NS_PER_US		1000UL
-#define HIST_MAX_ENTRIES	1000
+#define NSEC_PER_SEC		1000000000UL
+#define NSEC_PER_USEC		1000UL
 
 struct stats {
 	pthread_t pid;
 	pid_t tid;
 	unsigned int affinity;
-	unsigned int max;	/* in us */
-	unsigned int min;	/* in us */
-	unsigned int avg;	/* in us */
+	unsigned int max;	/* in latency unit (us or ns) */
```

Comment:
Couldn't we do all measurements in ns only do the conversion to us when dumping the data? 
Maybe we need to update the data type anyway. 'unsigned int' could only be 32 bit  and  that is 4 seconds only.

Reply:
Yes, we could. But that would extremely increase the bucket size by a factor of 1000 for measurements with µs resolution. One entry in the bucket represents either 1ns, or 1us.

Reply:
Good point, though I still think we should change the code to do all the calculation in ns precession on default. The histogram could be then be either in us or ns resolution. Maybe we should make the bucket size independent.  Also the display functions should format the output on the fly. 

So we split the low level code from the report code how it handles the format. 

What do you think?